src/settings.js

```diff
@@ -118,6 +118,15 @@ class Settings extends SafeEventEmitter {
       const flagsToAdd = setFlag(defaultFlags, oldChangedBits, false);
       if (flagsToAdd > 0) {
         this.set(flagSettingId, setFlag(oldFlags, flagsToAdd, true));
+
+        // upgrade sidebar flags : split RECOMMENDED_CHANNELS option into one option per section
+        if (version <= '7.5.3' && flagSettingId === SettingIds.SIDEBAR) {
```

Comment:
does this string comparison work? you may need to use a semver comparison library?